internal/logger.go

```diff
@@ -16,36 +16,39 @@ package containersuseconnect
 
 import (
 	"fmt"
+	"io"
 	"log"
 	"os"
 )
 
 // The default path for the logger if nothing has been specified.
-var defaultLogPath = "/var/log/suseconnect.log"
-
-// The environment variable used to specify a custom path for the logger
-// path.
-const logEnv = "SUSECONNECT_LOG_FILE"
-
-// GetLoggerFile returns the output file for the logger. If the `logEnv` environment
-// variable has been set, it will try to output there. Otherwise, it
-// will try to output to the file as given in `defaultLogPath`. If
-// everything fails, it will just output to the standard error channel.
-func GetLoggerFile() *os.File {
-	// Determine the path to be used.
-	var path string
-	if env := os.Getenv(logEnv); env != "" {
-		path = env
-	} else {
-		path = defaultLogPath
+const DefaultLogPath = "/var/log/suseconnect.log"
+
+// The environment variable used to specify a custom path for the log file.
+const LogEnv = "SUSECONNECT_LOG_FILE"
+
+// GetLogPath returns the log file path. If the `LogEnv` environment
+// variable has been set, it will return its value if not empty.
+// Otherwise, it return `DefaultLogPath`.
+func GetLogPath() string {
+	if env := os.Getenv(LogEnv); env != "" {
+		return env
 	}
 
-	// If it's writable, use the given file, otherwise use os.Stderr.
-	f, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0640)
+	return DefaultLogPath
+}
```

Comment:
I liked the idea of guaranteeing that the file is writable. I reworked this section.

The only issue is that `/var/log` is not accessible for most users, if you invoke container-suseconnect it will warn that it can't write to it. The only solution for this problem would be to write logs somewhere else where all users can write to, but the only place they can might be /tmp or their home folder.